gammapy/image/maps.py

```diff
@@ -387,6 +387,48 @@ def cutout(self, position, size):
         skymap = SkyMap(data=cutout.data, wcs=cutout.wcs, unit=self.unit)
         return skymap
 
+    def pad(self, factor, mode, **kwargs):
+        """Pad image to a larger shape.
+
+        Calls `numpy.pad`, passing ``mode`` and ``kwargs`` to it.
+
+        Parameters
+        ----------
+        factor : int
+            Factor used for output shape computation
+        mode : str
+            Padding mode, passed to `numpy.pad`.
+
+        Returns
+        -------
+        image : `~gammapy.image.SkyMap`
+            Padded image
+
+        Examples
+        --------
+
+        >>> from gammapy.image import SkyMap
+        >>> image = SkyMap.empty(nxpix=10, nypix=13)
+        >>> print(image.data.shape)
+        (13, 10)
+        >>> image2 = image.pad(factor=4, mode='reflect')
+        >>> image2.data.shape
+        (16, 12)
+        """
+        pad_width = factor - (np.array(self.data.shape) % factor)
+        pad_width = [(0, pad_width[0]), (0, pad_width[1])]
```

Comment:
Thinking a bit more about this, I still prefer symmetrical padding (i.e. on all edges), so that the center of the image is unchanged (or at least add this as an option). The down- and upsampling can create artefacts at the data edges, depending on the boundary handling, that is chosen. I would find it weird, if this only affected the right and upper edge of the image. I would rather expect a consistent treatment of all the edges, e.g. how it's done for convolution operations.